crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -134,6 +135,53 @@ impl<'a> Checker<'a> {
         }
         noqa::rule_is_ignored(code, offset, self.noqa_line_for, self.locator)
     }
+
+    /// Create a [`Generator`] to generate source code based on the current AST state.
+    pub fn generator(&self) -> Generator {
```

Comment:
I don't know where this belongs -- it requires the `Context`, `Indexer`, `Locator`, and `Stylist`, so right now it lives on `Checker`...